### PR TITLE
Added leetcode solution for Problem number 1305

### PR DESCRIPTION
**Pull request by UtkarshBirla28**
## Related Issue

- [ ] Fix; #2779 
- [ ] Fix: #2780

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by UtkarshBirla28** [comment]
@sanjay-kv Sir i have solved 2 issuees in this PR both of level 1 so please give me 20 pointd for that :)

**Comment by sanjay-kv** [comment]
@UtkarshBirla28 anyhow this repo has limitation of 150 points giving 20 will reduce the no of PR you can issue. 
anyhow admin has to decide on this.

File: dsa-problems/leetcode-problems/1300-1399.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const problems = [
     "problemName": "1305. All Elements in Two Binary Search Trees",
     "difficulty": "Medium",
     "leetCodeLink": "https://leetcode.com/problems/all-elements-in-two-binary-search-trees",
-    "solutionLink": "#"
+    "solutionLink":  "/dsa-solutions/lc-solutions/1300-1399/1305-All- Elements- in- Two- Binary -Search -Trees"
```
Comment (Ajay-Dhangar) [inline dsa-problems/leetcode-problems/1300-1399.md:47]: wrong url

File: dsa-solutions/lc-solutions/1300-1399/1305- All- Elements- in- Two- Binary -Search -Trees.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+---
+id: All- Elements- in- Two- Binary -Search -Trees
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1300-1399/1305- All- Elements- in- Two- Binary -Search -Trees.md:2]: replace `All- Elements- in- Two- Binary -Search -Trees` to `all-elements-in-two-binary-search-trees`

File: dsa-solutions/lc-solutions/1300-1399/1305- All- Elements- in- Two- Binary -Search -Trees.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+title: All Elements in Two Binary Search Trees
+sidebar_label: 1305- All- Elements- in- Two- Binary -Search -Trees
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1300-1399/1305- All- Elements- in- Two- Binary -Search -Trees.md:4]: replace `1305- All- Elements- in- Two- Binary -Search -Trees` to `1305- All Elements in Two Binary Search Trees`

File: dsa-solutions/lc-solutions/1300-1399/1310 XOR-Queries-Of-A-Subarray.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+---
+id: XOR-Queries-Of-A-Subarray.
+title: XOR Queries Of A Subarray.
+sidebar_label: 1310 XOR-Queries-Of-A-Subarray
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1300-1399/1310 XOR-Queries-Of-A-Subarray.md:4]: replace `1310 XOR-Queries-Of-A-Subarray` to `1310 XOR Queries Of A Subarray`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Review by Ajay-Dhangar — APPROVED** [review]
Debug the code to correct syntax errors.

**Pull request merged**